### PR TITLE
feature: refactor get the snippet code

### PR DESCRIPTION
**Pull request by xxxigm**
Changes made:

1. Removed unnecessary nested promises by chaining then() calls.
2. Used forEach loop instead of for loop for iterating over codeLines.
3. Replaced indexOf with includes for checking if a line contains the search term.
4. Simplified the logic for setting startingLine and foundLine within the loop.
5. Replaced arrow functions with concise arrow functions where possible.

### CODE REVIEW

File: components/contentComponents/playgroundSearchResult.tsx [hunk 1]
```diff
@@ -103,32 +103,35 @@ export const PlaygroundSearchResult: FunctionComponent<{ searchResult: IPlaygrou
...
-                            codeLines[foundLine] = `${codeLines[foundLine].substring(0, 100)}...`;
-                        }
-                        if (codeLines[foundLine].toLowerCase().indexOf(lowerTerm) !== -1) {
-                            startingLine = Math.max(foundLine - 5, 0);
-                            break;
-                        }
+        fetch(`https://snippet.babylonjs.com/${searchResult.snippetIdentifier}/${searchResult.version}`)
+        .then(response => response.text())
+        .then(text => {
+            const parsed = JSON.parse(JSON.parse(text).jsonPayload);
```
Comment (yedpodtrzitko) [inline components/contentComponents/playgroundSearchResult.tsx:109]: would this work?

```suggestion
        .then(response => response.json())
        .then(jsonData => {
            const parsed = JSON.parse(jsonData.jsonPayload);
```

Comment (xxxigm) [reply components/contentComponents/playgroundSearchResult.tsx:109]: @yedpodtrzitko I think keeping my version will work fine. As for yours, I'm not sure.

File: components/contentComponents/playgroundSearchResult.tsx [hunk 2]
```diff
@@ -103,32 +103,35 @@ export const PlaygroundSearchResult: FunctionComponent<{ searchResult: IPlaygrou
...
+            const lowerTerm = term.toLowerCase();
+    
+            if (type === "code") {
+                codeLines.forEach((line, index) => {
+                    if (line.length > 100) {
+                        codeLines[index] = `${line.substring(0, 100)}...`;
+                    }
+                    if (line.toLowerCase().includes(lowerTerm)) {
+                        foundLine = index;
+                        startingLine = Math.max(foundLine - 5, 0);
```
Comment (sebavan) [inline components/contentComponents/playgroundSearchResult.tsx:121]: Without the break, the code might not early exit anymore, why this change ?

Comment (xxxigm) [reply components/contentComponents/playgroundSearchResult.tsx:121]: @sebavan Wow, thank you for your question. Updated!

File: components/contentComponents/playgroundSearchResult.tsx (hunk 1 above)
Comment (sebavan) [reply components/contentComponents/playgroundSearchResult.tsx:109]: We should try as it is better ? can you check @NguyenTuanCanh ?

File: components/contentComponents/playgroundSearchResult.tsx (hunk 2 above)
Comment (sebavan) [reply components/contentComponents/playgroundSearchResult.tsx:121]: Your version is still different as foundLine will not be set outside of the condition but was always set in the loop version. I would in this case recommend to keep the loop as there is no reasons not too ?

File: components/contentComponents/playgroundSearchResult.tsx (hunk 1 above)
Comment (RaananW) [reply components/contentComponents/playgroundSearchResult.tsx:109]: Both should work, one needs one less JSON.parse call.

**Comment by RaananW** [comment]
This is mainly a cosmetic change, right? No functionality has changed?

**Comment by RaananW** [comment]
I am a bit confused as to the current changes. Formatting has changed a bit, but it seems like the code as it is is almost (or even completely) the same?

**Comment by RaananW** [comment]
any update here? or should I close it?
